pages/1.13/installing/production/system-requirements/zookeeper-requirements/index.md

```diff
@@ -0,0 +1,62 @@
+One key way you can identify issues that are related to ZooKeeper is by searching DC/OS exhibitor (`dcos-exhibitor`) log files for messages related to synchronization (`fsync`) operations. For example, if the ZooKeeper data and log directories are mounted on the same disk or there are disk or network latency issues, the transaction log records messages similar to the following:
+
+`WARN SyncThread:14  fsync-ing the write ahead log in SyncThread:14 took 14818ms which will adversely effect operation latency. See the ZooKeeper troubleshooting guide`
+
+# Recommendations for a healthy DC/OS cluster
+Because ZooKeeper keeps track of state, it is sensitive to timeouts caused by network latency. If you experience issues with network bandwidth being overloaded or with client sessions that are terminated because network connections are too slow, these issues will also make your DC/OS cluster less reliable.
+
+To ensure you have a healthy DC/OS cluster, you should use the following guidelines to deploy and manage ZooKeeper.
+
+## Monitor memory and swap
+You should verify that Zookeeper has enough heap memory. Insufficient memory allocation can affect ZooKeeper performance particularly during garbage collection. Specific memory requirements, however, vary depending on the number of nodes, clients, and schedulers you have deployed for the cluster, and the overall cluster workload you need to run. 
+
+In most cases, you should also configure your ZooKeeper installation to not allow memory swapping. For ZooKeeper operations to function correctly in a timely way, you should avoid allowing swap space to be used. The maximum heap size you allocate for ZooKeeper should not be larger than the real memory available for to ZooKeeper to use.
+
+One simple way you can monitor the read and write performance of a disk device is by running the following commands to measure server throughput and server latency:
+<p>
+<code>dd if=/dev/zero of=/tmp/test1.img bs=1024 count=1</code><br>
+<code>dd if=/dev/zero of=/tmp/test2.img bs=1024 count=1000</code>
+
+## Isolate master nodes from agent nodes
+
+Isolate master nodes from agent nodes to prevent memory and CPU contention, especially when workloads are started on the agents.
+
+The most common ZooKeeper issues occur when ZooKeeper transaction log files are stored on the same solid state drive (SSD) as other components where ZooKeeper write performance becomes a bottleneck that slows down or delays processing of other worker threads.
+    
+ZooKeeper should not share resources with any other processes or services. ZooKeeper writes new transactions to the log before performing updates or sending a response to the client. Dedicating disk space for the ZooKeeper transaction log directory prevents I/O processing by other applications from overloading the disk.
+    
+You should be sure ZooKeeper directories are configured to use fast disks that can complete synchronization (`fsync`) operations successfully in a timely fashion.
```

Comment:
I suggest moving the potential problems text here as:

If there are disk latency issues writing to the transaction log, ZooKeeper may log messages similar to the following:

`WARN SyncThread:14  fsync-ing the write ahead log in SyncThread:14 took 14818ms which will adversely effect operation latency. See the ZooKeeper troubleshooting guide`

(Note I have removed network latency - this is only relevant if the disk if network mounted - that is, it is a specific way in which a disk might have high latency).

Reply:
changed.